File: panaderia-backend/src/main/java/com/besoft/panaderia/service/ComprobanteService.java
package com.besoft.panaderia.service;

import java.util.List;

import com.besoft.panaderia.dto.request.ComprobanteBuscarRequest;
import com.besoft.panaderia.dto.request.ComprobanteRequest;
import com.besoft.panaderia.dto.response.OutResponse;
import com.besoft.panaderia.dto.response.ComprobanteResponse;

public interface ComprobanteService {

	public OutResponse<List<ComprobanteResponse>> listarComprobante(ComprobanteBuscarRequest req);

	public OutResponse<ComprobanteResponse> registrarComprobante(ComprobanteRequest req);

	public OutResponse<ComprobanteResponse> modificarComprobante(ComprobanteRequest req);

	public OutResponse<ComprobanteResponse> eliminarComprobante(ComprobanteRequest req);
	
	public OutResponse<ComprobanteResponse> establecerComprobanteActual(ComprobanteRequest req);
}

File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/comprobante.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { ComprobanteBuscarRequest } from '../dto/request/comprobante-buscar.request';
import { ComprobanteRequest } from '../dto/request/comprobante.request';
import { ComprobanteResponse } from '../dto/response/comprobante.response';
import { OutResponse } from '../dto/response/out.response';

@Injectable()
export class ComprobanteService {

  constructor(private http: HttpClient) { }

  public listarComprobante(req: ComprobanteBuscarRequest): Observable<OutResponse<ComprobanteResponse[]>> {
    return this.http.post<OutResponse<ComprobanteResponse[]>>(`${environment.WsPanaderiaBackend}/comprobante/listarComprobante`, req);
  }

  public registrarComprobante(req: ComprobanteRequest): Observable<OutResponse<ComprobanteResponse>> {
    return this.http.post<OutResponse<ComprobanteResponse>>(`${environment.WsPanaderiaBackend}/comprobante/registrarComprobante`, req);
  }

  public modificarComprobante(req: ComprobanteRequest): Observable<OutResponse<ComprobanteResponse>> {
    return this.http.post<OutResponse<ComprobanteResponse>>(`${environment.WsPanaderiaBackend}/comprobante/modificarComprobante`, req);
  }

  public eliminarComprobante(req: ComprobanteRequest): Observable<OutResponse<ComprobanteResponse>> {
    return this.http.post<OutResponse<ComprobanteResponse>>(`${environment.WsPanaderiaBackend}/comprobante/eliminarComprobante`, req);
  }
}
